tests/Unit/Domain/Test_Block.cpp

```diff
@@ -17,52 +18,143 @@
+  test_serialization(original_block);
 
   // Test move semantics:
   const Block<Dim> block_copy(identity_map.get_clone(), 7, {});
-  test_move_semantics(std::move(block), block_copy);
+  test_move_semantics(std::move(original_block), block_copy);
+}
+
+template <size_t Dim>
+void test_block_time_dependent() {
+  using Translation = domain::CoordMapsTimeDependent::Translation;
+  using TranslationDimD = tmpl::conditional_t<
+      Dim == 1,
+      domain::CoordinateMap<Frame::Grid, Frame::Inertial, Translation>,
+      tmpl::conditional_t<
+          Dim == 2,
+          domain::CoordinateMap<Frame::Grid, Frame::Inertial,
+                                domain::CoordMapsTimeDependent::ProductOf2Maps<
+                                    Translation, Translation>>,
+          domain::CoordinateMap<Frame::Grid, Frame::Inertial,
+                                domain::CoordMapsTimeDependent::ProductOf3Maps<
+                                    Translation, Translation, Translation>>>>;
+  using logical_to_grid_coordinate_map =
+      CoordinateMap<Frame::Logical, Frame::Inertial,
+                    CoordinateMaps::Identity<Dim>>;
+  using grid_to_inertial_coordinate_map = TranslationDimD;
+  PUPable_reg(logical_to_grid_coordinate_map);
+  PUPable_reg(SINGLE_ARG(CoordinateMap<Frame::Logical, Frame::Grid,
+                                       CoordinateMaps::Identity<Dim>>));
+  PUPable_reg(grid_to_inertial_coordinate_map);
+  const logical_to_grid_coordinate_map identity_map{
+      CoordinateMaps::Identity<Dim>{}};
+  const grid_to_inertial_coordinate_map translation_map{TranslationDimD{}};
+  Block<Dim> original_block(identity_map.get_clone(), 7, {});
+  CHECK_FALSE(original_block.is_time_dependent());
+  original_block.inject_time_dependent_map(translation_map.get_clone());
+  CHECK(original_block.is_time_dependent());
+
+  const auto check_block = [](const Block<Dim>& block) {
+    const double time = 2.0;
+
+    std::unordered_map<std::string,
+                       std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+        functions_of_time{};
+
+    functions_of_time["trans"] =
+        std::make_unique<FunctionsOfTime::PiecewisePolynomial<2>>(
+            0.0, std::array<DataVector, 3>{{{0.0}, {1.0}, {0.0}}});
+
+    // Test external boundaries:
+    CHECK((block.external_boundaries().size()) == 2 * Dim);
+
+    // Test neighbors:
+    CHECK((block.neighbors().size()) == 0);
+
+    // Test id:
+    CHECK((block.id()) == 7);
+
+    // Test that the block's coordinate_map is Identity:
+    const auto& grid_to_inertial_map = block.moving_mesh_grid_to_inertial_map();
+    const auto& logical_to_grid_map = block.moving_mesh_logical_to_grid_map();
+    const tnsr::I<double, Dim, Frame::Logical> xi(1.0);
+    const tnsr::I<double, Dim, Frame::Inertial> x(1.0 + time);
+    CHECK(grid_to_inertial_map(logical_to_grid_map(xi), time,
+                               functions_of_time) == x);
+    CHECK(
+        logical_to_grid_map
```

Comment:
It seems like it will be a pretty common use case to want the composition of maps from the logical all the way to the inertial frame as is effectively done here. Should there be utilities in the block to perform the composed operation, or is there a clear reason that those should not be provided?

Reply:
In general I think we don't really want to combine the maps (and I don't think it'll occur very often) because we want to cache the grid quantities since they're time-independent. Then we only need to compute the maps from grid->inertial. I think currently only the Minmod limiter calls the maps directly, everything else just needs the element's grid points